src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PhysicalFileProvider.cs

```diff
@@ -177,22 +178,28 @@ private void ReadPollingEnvironmentVariables()
         /// <summary>
         /// Disposes the provider. Change tokens may not trigger after the provider is disposed.
         /// </summary>
-        public void Dispose() => Dispose(true);
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
```

Comment:
`GC.SuppressFinalize(this);` is not necessary on types that don't have finalizers.  You removed the finalizer below.

Reply:
This was discussed in https://github.com/dotnet/runtime/pull/41918#discussion_r485941725